### PR TITLE
Add `mdast-util-to-string` as a dependency

### PR DESCRIPTION
**Pull request by wooorm**
...and remove `remark`, `strip-markdown`.

Reason for this is that it’s a lot faster to just stringify the `item`.
Also: there weren’t any tests by I checked it on:

```md
*   foo
*   bar
*   baz
```

...and the result were still OK :smile:

### CODE REVIEW

**Comment by vhf** [comment]
Thanks, I'll merge and test. I was wondering, will you also rename `mdast-util-position` and `mdast-util-to-string` to `remark-` or not?

**Comment by wooorm** [comment]
Nope, they won’t because they work with the AST (which keeps the moniker “mdast”)

File: lib/alphabetical-list-items.js [hunk 1]
```diff
@@ -22,9 +20,7 @@ function alphaCheck(ast, file, language, done) {
 
     items.forEach((item) => {
       if (item.children.length) {
-        const lineStartOffset = item.children[0].children[0].position.start.offset;
-        const lineEndOffset = item.children[0].children[item.children[0].children.length - 1].position.end.offset;
-        const text = normalize(remark.process(contents.slice(lineStartOffset, lineEndOffset)));
+        const text = normalize(toString(item));
```
Comment (vhf) [inline lib/alphabetical-list-items.js]: I think this converts the item and its subnodes to text. The original code only converts the line on which the listItem is found to string.

Comment (wooorm) [inline lib/alphabetical-list-items.js]: Yup, why would that be a problem for comparison?

Comment (vhf) [inline lib/alphabetical-list-items.js]: Take a look at the following Markdown list:

```
* a
  * z
* a a
```

I consider it sorted. If we "toString" the whole `a` node, `a z` will be compared to `a a`, and this linter plugin will suggest to swap `a` with `a a`. :)

Comment (wooorm) [inline lib/alphabetical-list-items.js]: Oh that’s an interesting case, I was thinking that it would be better to sort it by indeed preferring `a a, a z`.

I like your solution of using `item.children[0].children[0]`, but it might pose fragile for cases like empty list-items, if you’re adding tests I suggest inserting empty list-items, and list-items with just a code block too!

Comment (vhf) [inline lib/alphabetical-list-items.js]: I'll add tests in the next few days. Until now I have been testing with the whole free-programming-books repo.
